samples/apps/bookinfo/src/build-docker-services.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+#
+# Copyright 2017 Istio Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+# Set required env vars. Ensure you have checked out the pilot project
+SCRIPTDIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+DOCKER_REPO_ID=linsun
+WORKSPACE=$GOPATH/src/istio.io/pilot
+BINDIR=$WORKSPACE/bazel-bin
+APPSDIR=$SCRIPTDIR
+DISCOVERYDIR=$SCRIPTDIR/../../discovery
+PILOTAGENTPATH=$WORKSPACE/cmd/pilot-agent
+PILOTDISCOVERYPATH=$WORKSPACE/cmd/pilot-discovery
+PREPAREPROXYSCRIPT=$WORKSPACE/docker
+
+# grab ISTIO_PROXY_BUCKET from pilot/WORKSPACE
+ISTIO_PROXY_BUCKET=$(sed 's/ = /=/' <<< $( awk '/ISTIO_PROXY_BUCKET =/' $WORKSPACE/WORKSPACE))
+PROXYVERSION=$(sed 's/[^"]*"\([^"]*\)".*/\1/' <<<  $ISTIO_PROXY_BUCKET)
+# configure whether you want debug or not
+PROXY=debug-$PROXYVERSION
+echo Using proxy version $PROXY
+
+set -x
+set -o errexit
+
+# Build the pilot agent binary
+cd $PILOTAGENTPATH && bazel build :pilot-agent
+STATUS=$?
+if [ $STATUS -ne 0 ]; then
+    echo -e "\n***********\nFAILED: build failed for pilot agent.\n***********\n"
+    exit $STATUS
+fi
+
+# Build the pilot discovery binary
+cd $PILOTDISCOVERYPATH && bazel build :pilot-discovery
+STATUS=$?
+if [ $STATUS -ne 0 ]; then
+    echo -e "\n***********\nFAILED: build failed for pilot discovery.\n***********\n"
+    exit $STATUS
+fi
+
+cd $DISCOVERYDIR
+rm -f pilot-discovery && cp $BINDIR/cmd/pilot-discovery/pilot-discovery $_
+docker build -t $DOCKER_REPO_ID/discovery:latest .
```

Comment:
Also across istio it is common to have TAG and HUB env variables.

Reply:
ok fair point!